ESLint 4.0 alpha: False positive for space-before-function-paren

**Tell us about your environment**

* **ESLint Version:** v4.0.0-alpha.0
* **Node Version:** v7.8.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```
env:
  node: true
  es6: true

parserOptions:
  ecmaVersion: 2017

rules:
  space-before-function-paren: [2, never]
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
'use strict';
const assert = require('assert');
const util = require('util');

assert.strictEqual(util.inspect(async () => {}), '[AsyncFunction]');
```

**What did you expect to happen?**
I expected no errors to be reported because the docs say that async arrow functions are ignored by default.

**What actually happened? Please include the actual, raw output from ESLint.**

Got this instead:

```
  5:38  error  Unexpected space before function parentheses  space-before-function-paren
```

--- a/docs/rules/space-before-function-paren.md
+++ b/docs/rules/space-before-function-paren.md
@@ -33,7 +33,7 @@ This rule has a string option or an object option:
     "space-before-function-paren": ["error", {
         "anonymous": "always",
         "named": "always",
-        "asyncArrow": "ignore"
+        "asyncArrow": "always"
     }],
 }
 ```
@@ -44,13 +44,11 @@ This rule has a string option or an object option:
 The string option does not check async arrow function expressions for backward compatibility.
 
 You can also use a separate option for each type of function.
-Each of the following options can be set to `"always"`, `"never"`, or `"ignore"`.
-Default is `"always"` basically.
+Each of the following options can be set to `"always"`, `"never"`, or `"ignore"`. The default is `"always"`.
 
 * `anonymous` is for anonymous function expressions (e.g. `function () {}`).
 * `named` is for named function expressions (e.g. `function foo () {}`).
 * `asyncArrow` is for async arrow function expressions (e.g. `async () => {}`).
-  `asyncArrow` is set to `"ignore"` by default for backwards compatibility.
 
 ### "always"
 
@@ -83,6 +81,8 @@ var foo = {
         // ...
     }
 };
+
+var foo = async() => 1
 ```
 
 Examples of **correct** code for this rule with the default `"always"` option:
@@ -115,9 +115,7 @@ var foo = {
     }
 };
 
-// async arrow function expressions are ignored by default.
 var foo = async () => 1
-var foo = async() => 1
 ```
 
 ### "never"
@@ -151,6 +149,8 @@ var foo = {
         // ...
     }
 };
+
+var foo = async () => 1
 ```
 
 Examples of **correct** code for this rule with the `"never"` option:
@@ -183,8 +183,6 @@ var foo = {
     }
 };
 
-// async arrow function expressions are ignored by default.
-var foo = async () => 1
 var foo = async() => 1
 ```
 

